environment.cpp

```diff
@@ -7,6 +7,7 @@ environment::environment(double target_valueA, double target_valueB, std::functi
     m_ref_target_values{target_valueA,target_valueB},
     m_current_target_value {target_valueA},
     m_input(3, 0.5),
+    m_optimal_output{1},
```

Comment:
I might be wrong, but maybe there is no need to initialize this memeber at the creation of the object `environment`, in any case the optimal value is gonna be determined for each timestep, so already the first time tick is called an optimal value is going to be assigned.

Reply:
This caused failures in some old tests that checked equality between simulations when it was not in there. Because the equality was tested in the test without any tick being done, so no optimal value was assigned in there. And without a default value two newly-initialized environments were actually not equal in that context.